src/o365/spfx/commands/project/project-externalize/rules/PnPJsRule.ts

```diff
@@ -40,20 +45,52 @@ export class PnPJsRule extends BasicDependencyRule {
     }
   ];
 
-  public visit(project: Project): Promise<ExternalizeEntry[]> {
+  public visit(project: Project): Promise<[ExternalizeEntry[],FileEditSuggestion[]]> {
     const findings = this.pnpModules
       .map(x => this.getModuleAndParents(project, x.key))
       .reduce((x, y) => [...x, ...y]);
-
-    if (findings.filter(x => x.key !== '@pnp/pnpjs').length > 0) {
+    const files = this.getEntryFilesList(project);
```

Comment:
It would be helpful to add some comments in this section to simplify maintenance. Perhaps it will get simplified when we move from array to an object.

Reply:
done